src/Java/gtPlusPlus/core/material/Material.java

```diff
@@ -725,10 +726,19 @@ public final boolean requiresBlastFurnace(){
 	public final ItemStack getComponentByPrefix(OrePrefixes aPrefix, int stacksize) {
 		String aKey = aPrefix.name();
 		String aOreName = aKey + this.unlocalizedName;
-		for (ItemStack tTarget : OreDictionary.getOres(aOreName)) {
-			if ("bartworks".equals(ItemUtils.getModId(tTarget)) || "gregtech".equals(ItemUtils.getModId(tTarget)))
-				return GT_Utility.copyAmount(stacksize, tTarget);
+
+		if (mOreDictItemCache.containsKey(aOreName)) {
+			return GT_Utility.copyAmount(stacksize, mOreDictItemCache.get(aOreName));
+		}
+		else {
+			for (ItemStack tTarget : OreDictionary.getOres(aOreName)) {
+				if ("bartworks".equals(ItemUtils.getModId(tTarget)) || "gregtech".equals(ItemUtils.getModId(tTarget))) {
```

Comment:
Why specifically target BW or GT? Unification should handle this if and when required.

Reply:
it won't and then it will generate dupe recipe like 2 small pile dust -> 1 full dust or 1 ingot -> 1 block

Reply:
So it's a problem in recipe generation itself..?

Reply:
gt should has a recipe check when adding recipe, but it fails to check when gtpp and bw both exist. considering the bw material is the main stream in nh, so i made the change in gtpp to force it use bw items.

Reply:
Fix the issue at the source, don't try butcher up code here.
You seem to prefer bandaids over proper fixes or doing half assed jobs.

Reply:
source is here. and what i am doing is the proper fix.

Reply:
It's not, or the issue would occur with just GT & GT++.

> gt should has a recipe check when adding recipe, but it fails to check when gtpp and bw both exist.

You said yourself it's GT.